cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java

```diff
@@ -36,33 +38,42 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * Real-time sink for Streams
  */
+@Plugin(type = "sink")
+@Name("StreamSink")
+@Description("Real-time sink that outputs to the specified CDAP Stream")
 public class StreamSink extends RealtimeSink<StructuredRecord> {
   private static final Logger LOG = LoggerFactory.getLogger(StreamSink.class);
-  private String streamName;
-  private String headersField;
-  private String dataField;
 
-  @Override
-  public void configure(StageConfigurer configurer) {
-    configurer.setDescription("Real-time sink that outputs to the specified stream.");
-    configurer.addProperty(new Property(Properties.Stream.NAME,
-                                        "The name of the stream to output to. Must be a valid stream name. " +
-                                          "The stream will be created if it does not exist.",
-                                        true));
-    configurer.addProperty(new Property(Properties.Stream.DATA_FIELD,
-                                        "Name of the field in the record that contains the data to be written to " +
-                                          "the specified stream. The data could be in binary format as " +
-                                          "a byte array or a ByteBuffer. It can also be a String." +
-                                          "If unspecified, the 'data' key is used.",
-                                        false));
-    configurer.addProperty(new Property(Properties.Stream.HEADERS_FIELD,
-                                        "Name of the field in the record that contains headers. Headers are presumed" +
-                                          " to be a map of string to string.",
-                                        false));
+  private static final String NAME_DESC = "The name of the stream to output to. Must be a valid stream name. " +
+    "The stream will be created if it does not exist.";
+  private static final String DATA_FIELD_DESC = "Name of the field in the record that contains the data to be " +
+    "written to the specified stream. The data could be in binary format as a byte array or a ByteBuffer. " +
+    "It can also be a String. If unspecified, the 'data' key is used.";
+  private static final String HEADERS_FIELD_DESC = "Name of the field in the record that contains headers. " +
+    "Headers are presumed to be a map of string to string.";
+
+  private StreamConfig streamConfig;
```

Comment:
You can have this field `final` and have a constructor `StreamSink(StreamConfig config)` (you might need to turn `StreamConfig` class public for that.